tests/test_wait_signal.py

```diff
@@ -640,6 +641,332 @@ def test_signals_and_callbacks_length_mismatch(self, qtbot, signaller):
+        """
+        if qt_api.pytest_qt_api == 'pyside':
+            signal = (signaller.signal, "signal()")
+        else:
+            signal = signaller.signal
+
+        with pytest.raises(SignalTimeoutError) as excinfo:
+            with qtbot.waitSignal(signal=signal, timeout=200, check_params_cb=None, raising=True):
+                pass  # don't emit any signals
+        ex_msg = TestWaitSignalsSignalTimeoutErrorMessage.get_exception_message(excinfo)
+        assert ex_msg == "Signal signal() not emitted after 200 ms"
+
+    def test_with_single_arg(self, qtbot, signaller):
+        """
+        In a situation where a signal with one argument is expected but the emitted instances have values that are
+        rejected by a callback, tests that the SignalTimeoutError message contains the name of the signal and the
+        list of non-accepted arguments.
+        """
+        if qt_api.pytest_qt_api == 'pyside':
+            signal = (signaller.signal_single_arg, "signal_single_arg(int)")
+        else:
+            signal = signaller.signal_single_arg
+
+        def arg_validator(int_param):
+            return int_param == 1337
+
+        with pytest.raises(SignalTimeoutError) as excinfo:
+            with qtbot.waitSignal(signal=signal, timeout=200, check_params_cb=arg_validator, raising=True):
+                signaller.signal_single_arg.emit(1)
+                signaller.signal_single_arg.emit(2)
+        ex_msg = TestWaitSignalsSignalTimeoutErrorMessage.get_exception_message(excinfo)
+        assert ex_msg == "Signal signal_single_arg(int) emitted with parameters [1, 2] within 200 ms, " \
+                         "but did not satisfy the arg_validator callback"
+
+    def test_with_multiple_args(self, qtbot, signaller):
+        """
+        In a situation where a signal with two arguments is expected but the emitted instances have values that are
+        rejected by a callback, tests that the SignalTimeoutError message contains the name of the signal and the
+        list of tuples of the non-accepted arguments.
+        """
+        if qt_api.pytest_qt_api == 'pyside':
+            signal = (signaller.signal_args, "signal_args(QString,int)")
+        else:
+            signal = signaller.signal_args
+
+        def arg_validator(str_param, int_param):
+            return str_param == "1337" and int_param == 1337
+
+        with pytest.raises(SignalTimeoutError) as excinfo:
+            with qtbot.waitSignal(signal=signal, timeout=200, check_params_cb=arg_validator, raising=True):
+                signaller.signal_args.emit('1', 1)
+                signaller.signal_args.emit('2', 2)
+        ex_msg = TestWaitSignalsSignalTimeoutErrorMessage.get_exception_message(excinfo)
+        assert ex_msg == "Signal signal_args(QString,int) emitted with parameters [('1', 1), ('2', 2)] " \
+                         "within 200 ms, but did not satisfy the arg_validator callback"
```

Comment:
You can just use parens for line continuation instead of using `\` - I think `\` is too brittle, e.g. accidentally inserting a space after it suddenly won't do what you expect anymore.

(Same for other message checks in this file)